Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockAnalyzerInstaller.cs

```diff
@@ -29,6 +30,14 @@ namespace SonarScanner.MSBuild.PreProcessor.Test
 {
     internal class MockAnalyzerInstaller : IAnalyzerInstaller
     {
+        #region Constructors
+        public MockAnalyzerInstaller() { }
+
+        public MockAnalyzerInstaller(string localCacheDirectory) {
```

Comment:
This seems to be added to be called in the `MockObjectFactory.cs` and that does not use the installer. I would remove the constructors. 

The test itself checks that a folder exists (`AssertDirectoryExists(tmpCachePath);`), a folder that gets created by the unit tests. 

We could extend `MockObjectFactory` to log the call to `CreateRoslynAnalyzerProvider` with the parameters used and then assert that the correct path has been used. As you mentioned in another comment, the responsibility of creating the folder is on `EmbeddedAnalyzerInstaller`.

Reply:
@costin-zaharia-sonarsource So my initial feeling on this was aligned. Question then - is there an example in the codebase on how to log the path so I can assert it in the test later on?

Reply:
You can do something similar to what was done in `MockRoslynAnalyzerProvider`. It stores the `sonarProperties` as a class property in the method call.

Reply:
Thank you for that. Once I started looking at what happened there, this all made more sense. I think the tests are more inline with what you requested.

Reply:
Thank you!